include/mbedtls/asn1write.h

```diff
@@ -381,6 +362,31 @@ mbedtls_asn1_named_data *mbedtls_asn1_store_named_data(mbedtls_asn1_named_data *
                                                        const unsigned char *val,
                                                        size_t val_len);
 
+/**
+ * \brief                  Encode an integer into ASN1 and write it to a buffer.
+ *                         The function takes a big endian bignum buffer that
+ *                         begins at start and ends at *p and encodes it into
+ *                         a asn1 frame.
+ *                         Warning: Destructive behaviour for the buffer data!
+ *
+ * \param p                The reference to the current position pointer.
+ * \param start            The start of the buffer, for bounds-checking.
+ * \param integer          The start of the integer buffer.
```

Comment:
Is it ok if `integer` overlaps with the output buffer? That can be useful in typical cases that we'll encounter when printing out a bignum. For that we first convert the bignum to a big-endian byte array, then convert that to ASN.1. If we can do the second step in place, then we don't need to allocate an intermediate step.

We may not want to promise that it works if it's too difficult to implement. We do have to be careful that the ASN.1 representation is usually a few bytes larger than the byte array (because there's 2 or more bytes of overhead for the ASN.1 type, ASN.1 length and the possible additional byte for the ASN.1 sign bit), but it can also be smaller if the byte array has leading zeros.

Reply:
Currently overlap is not handled, but it should be possible using `memmove()`. I'll have a careful think about it tomorrow and see if it is easy to do.